src/ekat/ekat_workspace.hpp

```diff
@@ -131,6 +131,19 @@ class WorkspaceManager
     void take_many_contiguous_unsafe(const Kokkos::Array<const char*, N>& names,
                                      const view_1d_ptr_array<S, N>& ptrs) const;
 
+    // Take a block of size n*m_size, where m_size is the size used to construct
+    // the WorkspaceManager. This is useful for creating local 2d views through
+    // the WorkspaceManager.
+    //
+    // Example: Local 2d view of size (n, m_size).
+    // Code:
+    //   const auto local_slot = workspace.take_n_size_block("local_slot",n);
+    //   const auto local_var  = Unmanaged2dViewT<ScalarT>(reinterpret_cast<ScalarT*>(local_slot.data()),
+    //                                                     n, m_sizes);
+    template <typename S=T>
+    KOKKOS_INLINE_FUNCTION
+    Unmanaged<view_1d<S> > take_n_size_block(const char* name, const int n) const;
```

Comment:
The name of the method seems confusing. I guess it should be read as "take n size-sized blocks"? Or "take a NxSize block"?

How about something like `take_spliced_blocks`? Or another word (instead of splice) that conveys the msg that you are joining n blocks? `take_macro_block`? `take_super_block`?

Reply:
I like `take_macro_block`